backend/src/controllers/v2/secretController.ts

```diff
@@ -27,23 +31,44 @@ export const createSingleSecret = async (req: Request, res: Response) => {
     secretCommentTag: secretToCreate.secretCommentTag,
     secretCommentHash: secretToCreate.secretCommentHash,
     workspace: new Types.ObjectId(workspaceId),
-    environment: environmentName,
+    environment,
     type: secretToCreate.type,
     user: new Types.ObjectId(req.user._id)
   }
 
 
-  const [error, newlyCreatedSecret] = await to(Secret.create(sanitizedSecret).then())
+  const [error, secret] = await to(Secret.create(sanitizedSecret).then())
   if (error instanceof ValidationError) {
     throw RouteValidationError({ message: error.message, stack: error.stack })
   }
 
-  res.status(200).send()
+  if (postHogClient) {
+    postHogClient.capture({
+      event: 'secrets added',
+      distinctId: req.user.email,
+      properties: {
+        numberOfSecrets: 1,
+        workspaceId,
+        environment,
+        channel: req.headers?.['user-agent']?.toLowerCase().includes('mozilla') ? 'web' : 'cli',
+        userAgent: req.headers?.['user-agent']
+      }
+    });
+  } 
```

Comment:
This PostHog boilerplate takes a lot of space and not easy to change in bulk in the future. Is it possible to seperate to function and use that function?

Reply:
Hmmm interesting! How would you do it? Personally I'd say it's already pretty concise, though it does take a lot of space

Reply:
What about we create a function to get `distinctId`, `channel`, `userAgent` from `req` and `context` variable which extends the default. Something like this;
```ts
type PostHogEventData = {
    event: string
    distinctId: string
    properties: Record<any,any> & {
        channel?: "web"|"cli"
        userAgent?: string
    }
}

function capturePostHogEvent(eventName: string, req: Request, context: Partial<PostHogEventData>){
    if(!postHogClient) return;

    const postHogData = Object.assign({
        event: eventName,
        distinctId: req.user.email,
        properties: {
            channel: req.headers?.['user-agent']?.toLowerCase().includes('mozilla') ? 'web' : 'cli',
            userAgent: req.headers?.['user-agent']
        }
    }, context)

    postHogClient.capture(postHogData);
}

...

capturePostHogEvent('secrets_added', req, {
    properties: {
        numberOfSecrets: 1,
        workspaceId,
        environment
    }
})
```

Reply:
Agree with this!

I think we can add this as an extension to `/services/PostHogClient` in the backend. We can convert it to a class and have some useful methods on there like this one to capture events.

In the interest of time and un-blocking some people, we're going to merge this PR and add in this change in a separate PR. Feel free to submit a new PR with this modularization after the merge!